### PR TITLE
Docs: Clarify creating a contract example code

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Checklist
- [x] All tests are passing
- [x] README / documentation was extended, if necessary
- [x] Used meaningful commit messages

### Description

Closes https://github.com/ethereum/solidity/issues/2661

This PR attempts to clarify the example code by making code and comments easier to parse, and making the explanatory comments clearer as to what they do and that they are not 'real world' examples.

### CODE REVIEW

File: docs/contracts.rst [hunk 1]
```diff
@@ -56,14 +56,15 @@ This means that cyclic creation dependencies are impossible.
         // creator and the assigned name.
         constructor(bytes32 _name) public {
             // State variables are accessed via their name
-            // and not via e.g. this.owner. This also applies
+            // and not via e.g., `this.owner`. This also applies
             // to functions and especially in the constructors,
-            // you can only call them like that ("internally"),
+            // you can only call them "internally",
```
Comment (chriseth) [inline docs/contracts.rst]: At this point, people do not yet know what "calling internally" means, so I think we should somehow keep the connection. I see the problem, though that "like this" does not really clarify what "this" is :)

File: docs/contracts.rst [hunk 2]
```diff
@@ -80,10 +81,11 @@ This means that cyclic creation dependencies are impossible.
         function transfer(address newOwner) public {
             // Only the current owner can transfer the token.
             if (msg.sender != owner) return;
-            // We also want to ask the creator if the transfer
-            // is fine. Note that this calls a function of the
+
+            // We ask the creator contract if the transfer
+            // should proceed using a function of the `TokenCreator`
```
Comment (chriseth) [inline docs/contracts.rst]: confusion: proceed using the function or ask using the function?

File: docs/contracts.rst (hunk 1 above)
Comment (ChrisChinchilla) [reply docs/contracts.rst]: @chriseth How about "from functions declared in this code", "you can only call functions declared in this contract code", or something like that? I'd finesse that text but I _think_ that encompasses what you mean?

Comment (axic) [reply docs/contracts.rst]: Perhaps: generally you should not call them via `this.`, except in specific cases. More on that later.

Comment (ChrisChinchilla) [reply docs/contracts.rst]: @axic and @chriseth how about this change? I wish you could create links in code comments, but you can't :(

**Review by chriseth — APPROVED** [review]

**Pull request merged**